controller

```diff
@@ -0,0 +1,40 @@
+
+encryption_tutorial.php
+
+<?php
+
+class Encryption_Tutorial extends CI_Controller {
+
+public function __construct() {
+parent:: __construct();
+$this->load->helper('form');
+$this->load->library('encrypt');
+$this->load->library('form_validation');
+}
+
+public function index() {
+$this->load->view('show_form');
+}
+
+public function key_encoder() {
+
+$this->form_validation->set_rules('key', 'Message', 'trim|required|xss_clean');
```

Comment:
Can you make this key as readable from a secure file?

Reply:
say file will have secure key, you must read from that file

Reply:
no

Reply:
may i know why?